app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -146,4 +147,38 @@ class HomeFragmentPresenter @Inject constructor(
   fun onTopicSummaryClicked(topicSummary: TopicSummary) {
     routeToTopicListener.routeToTopic(topicSummary.topicId)
   }
+
+  private val ongoingStoryListSummaryResultLiveData: LiveData<AsyncResult<OngoingStoryList>> by lazy {
+    topicListController.getOngoingStoryList()
+  }
+
+  private fun subscribeToPromotedStoryList() {
+    getAssumedSuccessfulOngoingStoryList().observe(fragment, Observer<OngoingStoryList> {
+      if (it.recentStoryCount > 0) {
```

Comment:
Is this "if" check necessary? If recentStoryCount is empty, the for loop shouldn't do anything right? Same for below.

Reply:
Sorry, I didn't read the explanation correctly.

Reply:
No issues, I will keep this comment for now, and will solve this.